Obstacle.java

```diff
@@ -0,0 +1,6 @@
+public interface Obstacle {
```

Comment:
Would be better to make `Obstacle` a class, as there is no need for such an interface